src/tabular-api-surface.ts

```diff
@@ -224,6 +224,36 @@ Please use the format 'prezzy' or '${instance.name}/tables/prezzy'.`);
 
     rowKeys = options.keys || [];
 
+    /*
+    The following line of code sets the timeout if it was provided while
+    creating the client. This will be used to determine if the client should
+    retry on DEADLINE_EXCEEDED errors. Eventually, this will be handled
+    downstream in google-gax.
+     */
+    const timeout =
+      this.bigtable &&
+      this.bigtable.options &&
+      this.bigtable.options.BigtableClient &&
+      this.bigtable.options.BigtableClient.clientConfig &&
+      this.bigtable.options.BigtableClient.clientConfig.interfaces &&
+      this.bigtable.options.BigtableClient.clientConfig.interfaces[
+        'google.bigtable.v2.Bigtable'
+      ] &&
+      this.bigtable.options.BigtableClient.clientConfig.interfaces[
+        'google.bigtable.v2.Bigtable'
+      ].methods &&
+      this.bigtable.options.BigtableClient.clientConfig.interfaces[
+        'google.bigtable.v2.Bigtable'
+      ].methods['ReadRows'] &&
+      this.bigtable.options.BigtableClient.clientConfig.interfaces[
+        'google.bigtable.v2.Bigtable'
+      ].methods['ReadRows'].timeout_millis
+        ? this.bigtable.options.BigtableClient.clientConfig.interfaces[
+            'google.bigtable.v2.Bigtable'
+          ].methods['ReadRows'].timeout_millis
+        : undefined;
```

Comment:
This is very hard to read - can you use [optional chaining](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Operators/Optional_chaining) to simplify it? 

something like `this.bigtable?.options?.BigtableClient?.clientConfig?.interfaces[
            'google.bigtable.v2.Bigtable'
          ]?.methods['ReadRows']?.timeout_millis?`

(not sure how it works with the square bracket syntax but I think it can get rid of at least a few lines

Reply:
I just added optional chaining to shorten this.